nemo/core/classes/modelPT.py

```diff
@@ -1342,3 +1342,17 @@ def update_save_restore_connector(cls, save_restore_connector):
             cls._save_restore_connector = save_restore_connector
         else:
             setattr(cls, '_save_restore_connector', save_restore_connector)
+
+    def on_after_backward(self):
```

Comment:
Dont do this at ModelPT level, Megatron has different logic to handle this but calls super() which will inturn call this and cause issues. Push it up to ASRModel. Will need more discussion from team to put it into ModelPT.

Reply:
For now, I dropped it as it was not necessary for this model. It did not help this model. We can discuss this and add it in a separate PR.

Reply:
Moved it to ASRModel for now.